src/persist/src/sqlite.rs

```diff
@@ -98,18 +107,26 @@ impl Consensus for SqliteConsensus {
         _deadline: Instant,
         key: &str,
     ) -> Result<Option<VersionedData>, ExternalError> {
-        let conn = self.conn.lock().map_err(PersistError::from)?;
-        let mut stmt = conn.prepare(
-            "SELECT sequence_number, data FROM consensus
-                 WHERE shard = $shard ORDER BY sequence_number DESC LIMIT 1",
-        )?;
-        stmt.query_row(named_params! {"$shard": key}, |row| {
-            let seqno = row.get("sequence_number")?;
-            let data: Vec<_> = row.get("data")?;
-            Ok(VersionedData { seqno, data })
-        })
-        .optional()
-        .map_err(|e| e.into())
+        let shard = key.to_owned();
+        let conn = Arc::clone(&self.conn);
+        mz_ore::task::spawn_blocking(
+            || "sqlite::head",
+            move || {
+                let conn = conn.lock().map_err(PersistError::from)?;
+                const Q: &str = "
+SELECT sequence_number, data FROM consensus
+WHERE shard = $shard ORDER BY sequence_number DESC LIMIT 1";
+                let mut stmt = conn.prepare(Q)?;
```

Comment:
@ruchirK I didn't notice this in code review, curious now why this one is prepare and the others are all prepare_cached?

Reply:
I think previously there was some lifetime reason? I forget tbh. I changed this one to be `prepare_cached` and it seems to compile fine so most likely is fine to change